### PR TITLE
Topic/peer direct

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ibverbs-tests-pr/44/ for details. (Mellanox internal link)

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ibverbs-tests-pr/45/ for details. (Mellanox internal link)

File: tests/peer-direct/env.h [hunk 1]
```diff
@@ -667,6 +680,8 @@ class peerdirect_test : public testing::Test, public T::Base, public T::Aux {
 				sge[i].length = 8;
 				memcpy(ptr, &op->wr.qword_va.data, 8);
 				ptr += 8;
+			} else if (op->type == IBV_PEER_OP_FENCE) {
+				continue;
```
Comment (rleon) [inline tests/peer-direct/env.h]: You will skip i++ for this op->type

File: tests/peer-direct/smoke.cc [hunk 2]
```diff
@@ -133,6 +133,8 @@ TYPED_TEST(peerdirect_test, t5_poll_abort) {
 	EXEC(this->check_fin(RECEIVER, 4));
 }
 
+#if 1
```
Comment (rleon) [inline tests/peer-direct/smoke.cc]: ????

File: tests/peer-direct/env.h (hunk 1 above)
Comment (Artemy-Mellanox) [inline tests/peer-direct/env.h]: It's ok. I won't create wr for this op->type.

File: tests/peer-direct/smoke.cc (hunk 2 above)
Comment (Artemy-Mellanox) [inline tests/peer-direct/smoke.cc]: leftover

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ibverbs-tests-pr/46/ for details. (Mellanox internal link)

**Comment by rleon** [comment]
+1

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-libverbs-tests-SimX-pr/15/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-libverbs-tests-SimX-pr/16/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-libverbs-tests-SimX-pr/17/ for details (Mellanox internal link).

**Comment by Di0gen** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ibverbs-tests-pr/47/ for details. (Mellanox internal link)

**Pull request merged**